### PR TITLE
Move based on current position

### PR DESCRIPTION
**Pull request by brandonspangler1**
Drone moves relative to its current position rather than the starting position.

### CODE REVIEW

**Comment by jenniferolenchak** [comment]
Ideally rework to allow for continuous movement when motion buttons are held down

File: Application/AirSimControl.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from pathlib import Path
 import sys
 
+from threading import Timer
```
Comment (jenniferolenchak) [inline Application/AirSimControl.py]: Is this still needed?

File: Application/AirSimControl.py [hunk 2]
```diff
@@ -41,27 +43,28 @@ def initializeAirSimClient(self):
 
         print("AirSim Client Initialized")
 
-    def startDroneMovement(self, command):
+    def startDroneMovement(self, command, moveButtons):
```
Comment (jenniferolenchak) [inline Application/AirSimControl.py]: Is moveButtons parameter still needed here?

File: Application/AirSimControl.py [hunk 3]
```diff
@@ -41,27 +43,28 @@ def initializeAirSimClient(self):
...
+            self.client.moveByVelocityBodyFrameAsync(0-self.movementVelocity, 0, 0, 1)
         elif (command == 'rotate_right'):
-            self.client.moveByVelocityAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, abs(0-self.movementVelocity)))
+            self.client.moveByVelocityBodyFrameAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, abs(0-self.movementVelocity)))
         elif (command == 'rotate_left'):
-            self.client.moveByVelocityAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, -abs(0-self.movementVelocity)))
+            self.client.moveByVelocityBodyFrameAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, -abs(0-self.movementVelocity)))
 
-    def stopDroneMovement(self, command):
+    def stopDroneMovement(self, moveButtons):
```
Comment (jenniferolenchak) [inline Application/AirSimControl.py]: Is moveButtons parameter still needed here?

File: Application/AirSimControl.py [hunk 4]
```diff
@@ -41,27 +43,28 @@ def initializeAirSimClient(self):
...
         self.client.moveByVelocityAsync(0, 0, 0, 3)
 
+
```
Comment (jenniferolenchak) [inline Application/AirSimControl.py]: Remove extra space?

**Review by jenniferolenchak — CHANGES_REQUESTED** [review]

File: Application/AirSimControl.py [hunk 5]
```diff
@@ -41,27 +43,28 @@ def initializeAirSimClient(self):
...
         elif (command == 'rotate_right'):
-            self.client.moveByVelocityAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, abs(0-self.movementVelocity)))
+            self.client.moveByVelocityBodyFrameAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, abs(0-self.movementVelocity)))
         elif (command == 'rotate_left'):
-            self.client.moveByVelocityAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, -abs(0-self.movementVelocity)))
+            self.client.moveByVelocityBodyFrameAsync(0, 0, 0, 1, drivetrain=airsim.DrivetrainType.MaxDegreeOfFreedom,  yaw_mode=YawMode(True, -abs(0-self.movementVelocity)))
 
-    def stopDroneMovement(self, command):
+    def stopDroneMovement(self, moveButtons):
         self.client.moveByVelocityAsync(0, 0, 0, 3)
```
Comment (jenniferolenchak) [inline Application/AirSimControl.py:63]: Should this be 3?

**Review by jenniferolenchak — CHANGES_REQUESTED** [review]

File: Application/AirSimControl.py (hunk 1 above)
Comment (brandonspangler1) [reply Application/AirSimControl.py]: Removed

File: Application/AirSimControl.py (hunk 2 above)
Comment (brandonspangler1) [reply Application/AirSimControl.py]: removed

File: Application/AirSimControl.py (hunk 3 above)
Comment (brandonspangler1) [reply Application/AirSimControl.py]: reverted

File: Application/AirSimControl.py (hunk 4 above)
Comment (brandonspangler1) [reply Application/AirSimControl.py]: removed

File: Application/AirSimControl.py (hunk 5 above)
Comment (brandonspangler1) [reply Application/AirSimControl.py:63]: That third parameter is for duration so the method is stopping drone movement for 3 seconds. Git is telling me you were the last person to edit that line but I tried it at 1 and didn't see a difference. I am assuming on release of the button it stops the movement but then moving again overrides this so the drone is never stuck for three seconds and as such we should probably leave it at three.

**Review by jenniferolenchak — APPROVED** [review]

**Pull request merged**